pkg/pillar/cmd/README.md

```diff
@@ -0,0 +1,273 @@
+# Agent Services
+
+This directory contains agent services for pillar.
+
+Each subdirectory contains exactly one agent, whose name is the name of the directory.
+
+Look at [verifier](./verifier/) for an example of the implementation.
+
+## Design Pattern
+
+The design pattern for each agent is as follows. When creating a new agent, ensure
+that you follow the design pattern.
+
+The package name in the agent's directory is the same as the directory name.
+It should contain at least the following files:
+
+* `README.md` describing the agent and its purpose.
```

Comment:
Are we eventually going to get rid of https://github.com/lf-edge/eve/tree/master/pkg/pillar/docs? Currently the per-agent documentation is stored there.

Reply:
I didn't think about that. I could go either way: store docs about each service inside the service directory, or keep them in the directory https://github.com/lf-edge/eve/tree/master/pkg/pillar/docs

I have no preference, so it isn't really fundamental to this PR restructure.

What are your thoughts?

Reply:
I actually do not have preference either :)